src/widgets/NewExpiryDateInput.js

```diff
@@ -0,0 +1,174 @@
+ * a date to be committed to the underlying model until a valid date has
+ * been submitted.
+ *
+ * Similar to EditableCell, this component renders a Cell when disabled,
+ * a Text wrapped in a TouchableComponent when un-focused and a TextInput
+ * when focused - A workaround for react-native crashes when large numbers
+ * of TextInput components are rendered.
+ *
+ * @param {string|number} value The value to render in cell
+ * @param {string|number} rowKey Unique key associated to row cell is in
+ * @param {string|number} columnKey Unique key associated to column cell is in
+ * @param {bool} disabled If `true` will render a plain Cell element with no interaction
+ * @param {bool} isFocused If `false` will TouchableOpacity that dispatches a focusAction
+ *                         when pressed. When `true` will render a TextInput with focus
+ * @param {func} editAction Action creator for handling editing of this cell.
+ *                          `(newValue, rowKey, columnKey) => {...}`
+ * @param {func} focusAction Action creator for handling focusing of this cell.
+ *                          `(rowKey, columnKey) => {...}`
+ * @param {func} focusNextAction Action creator for handling focusing of this cell.
+ *                          `(rowKey, columnKey) => {...}`
+ * @param {func}  dispatch Reducer dispatch callback for handling actions
+ * @param {Number}  width Optional flex property to inject into styles.
+ * @param {Bool}  isLastCell Indicator for if this cell is the last
+ *                                   in a row. Removing the borderRight if true.
+ * @param {String}  placeholder String to display when the cell is empty.
+ * @
+ *
+ */
+export const NewExpiryDateInput = React.memo(
+  ({
+    value,
+    rowKey,
+    columnKey,
+    isDisabled,
+    isFocused,
+    editAction,
+    focusAction,
+    focusNextAction,
+    dispatch,
+    isLastCell,
+    width,
+    debug,
+    placeholder,
+  }) => {
+    if (debug) console.log(`- ExpiryTextInputCell: ${value}`);
+
+    // Customhook managing the editing of an expiry date to stay valid.
+    const [expiryDate, finalseExpiryDate, setExpiryDate] = useExpiryDateMask(
+      formatExpiryDate(value)
+    );
+
+    // Helpers controlling the submitting of the expiry date. Losing focus/submitting
+    // Handed similarly, but losing focus will not auto focus the next cell. Changes
+    // to the underlying model are not committed until a valid date is entered.
+    const finishEditingExpiryDate = () => {
+      finalseExpiryDate();
+      dispatch(editAction(parseExpiryDate(expiryDate), rowKey, columnKey));
+    };
+
+    const onSubmit = () => {
+      finishEditingExpiryDate();
+      dispatch(focusNextAction(focusNextAction(rowKey, columnKey)));
+    };
+
+    const usingPlaceholder = placeholder && !expiryDate;
+    const {
+      expiryBatchTextView,
+      expiryBatchViewStyle,
+      expiryBatchText,
+      expiryBatchPlaceholderText,
+    } = newDataTableStyles;
```

Comment:
I have a want to have this in the module scope. Worth noting the pattern of having `style` on the end isn't consistent.

Reply:
47765b3edf0099f158e8a67639733dfbcac51028